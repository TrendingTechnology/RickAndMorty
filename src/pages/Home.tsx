import React, { useState } from "react"
import Button from "../components/Button/Button"
import Table from "../components/AllCharsTable/AllCharsTable"
import useAllChars from "../hooks/useAllChars"
import Loader from "react-loader-spinner"

function Home() {
  const [page, setPage] = useState<number>(1)

  // Get all charachters data
  const { data, error, loading } = useAllChars(page)

  const nextPageHandler = () => {
    setPage((prev) => prev + 1)
  }
  const prevPageHandler = () => {
    setPage((prev) => {
      if (prev - 1 > 0) {
        return prev - 1
      } else {
        return prev
      }
    })
  }

  return (
    <div className="flex flex-col h-screen items-center justify-center bg-red-100">
      <div className="flex flex-col items-center">
        <h1 className="py-10 font-semibold text-2xl">
          Rick and Morty Charchters
        </h1>
        {error ? (
          <span>{error}</span>
        ) : loading ? (
          <Loader
            type="Puff"
            color="#00BFFF"
            height={100}
            width={100}
            timeout={3000} //3 secs
          />
        ) : (
          data && <Table data={data} />
        )}

        <div className="w-full flex items-center space-x-5 justify-between my-5">
          <Button
            variant="black"
            className="bg-red-400"
            onClick={prevPageHandler}
          >
            Prev
          </Button>
          <span id="page" className="text-xl ">
            Page: {page}
          </span>
          <Button
            variant="black"
            className="bg-blue-500"
            onClick={nextPageHandler}
          >
            Next
          </Button>
        </div>
      </div>
    </div>
  )
}

export default Home
