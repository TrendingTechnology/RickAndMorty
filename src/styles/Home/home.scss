.home-layout {
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgba(254, 226, 226, var(--tw-bg-opacity));

  &_container {
    display: flex;
    flex-direction: column;

    align-items: center;

    &_title {
      font-size: 1.5rem;
      line-height: 2rem;
      font-weight: bold;
      padding-top: 2.5rem;
      padding-bottom: 2.5rem;
    }
  }

  &_buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    padding-bottom: 1.25rem;
    margin-top: 1.25rem;
    margin-bottom: 1.25rem;
    span {
      font-size: 1.25rem /* 20px */;
      line-height: 1.75rem /* 28px */;
    }

    &--next {
      --tw-bg-opacity: 1;
      background-color: rgba(248, 113, 113, var(--tw-bg-opacity));
    }
  }
}
