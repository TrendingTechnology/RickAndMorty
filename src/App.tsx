import { Route, Switch } from "react-router"
import Charachter from "./pages/Charachter"
import Home from "./pages/Home"

function App() {
  return (
    <>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/charachter/:id" component={Charachter} />
      </Switch>
    </>
  )
}

export default App
