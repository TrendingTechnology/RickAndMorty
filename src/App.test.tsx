import { cleanup, render } from "@testing-library/react"
import App from "./App"
import Home from "./pages/Home"
import Enzyme from "enzyme"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import toJson from "enzyme-to-json"
import { Router } from "react-router-dom"
import { createMemoryHistory } from "history"

Enzyme.configure({ adapter: new Adapter() })
afterEach(cleanup)

describe("snapshot test", () => {
  const history = createMemoryHistory()

  it("snapshot test Home page", () => {
    const wrapper = Enzyme.shallow(<Home />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it("renders text in home page", () => {
    const { container } = render(
      <Router history={history}>
        <App />
      </Router>
    )

    expect(container.textContent).toMatch("Rick and Morty Charchters")
  })
})
