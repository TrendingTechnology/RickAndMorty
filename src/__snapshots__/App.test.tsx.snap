// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test snapshot test Home page 1`] = `
<div
  className="flex flex-col h-screen items-center justify-center bg-red-100"
>
  <div
    className="flex flex-col items-center"
  >
    <h1
      className="py-10 font-semibold text-2xl"
    >
      Rick and Morty Charchters
    </h1>
    <Loader
      className=""
      color="#00BFFF"
      height={100}
      style={Object {}}
      timeout={3000}
      type="Puff"
      visible={true}
      width={100}
    />
    <div
      className="w-full flex items-center space-x-5 justify-between my-5"
    >
      <Button
        className="bg-red-400"
        onClick={[Function]}
        variant="black"
      >
        Prev
      </Button>
      <span
        className="text-xl "
        id="page"
      >
        Page: 
        1
      </span>
      <Button
        className="bg-blue-500"
        onClick={[Function]}
        variant="black"
      >
        Next
      </Button>
    </div>
  </div>
</div>
`;
